Core/DISET/private/BaseClient.py

```diff
@@ -218,11 +219,50 @@ def __findServiceURL( self ):
         gLogger.debug( "Removing banned URL", "%s" % i )
         urls.remove( i )
         
-    sURL = List.randomize( urls )[0]
-
+    randUrls = List.randomize( urls ) 
+    sURL = randUrls[0]
+    
+    if len( self.__bannedUrls ) > 0 and self.__nbOfUrls > 2:  # when we have multiple services then we can have a situation 
+      # when two service are running on the same machine with different port...
+      
+      retVal = Network.splitURL( sURL )
+      nexturl = None
+      if retVal['OK']:
+        nexturl = retVal['Value']
+      
+      found = False
+      for i in self.__bannedUrls:
+        retVal = Network.splitURL( i )
+        if retVal['OK']:
+          bannedurl = retVal['Value']
+        else:
+          break
+        
+        if nexturl[1] == bannedurl[1]:
+          found = True
+          break
+      if found:
+        nexturl = self.__selectUrl( nexturl, randUrls[1:] )
+        if nexturl:  # an url found which is in different host
+          sURL = nexturl
     gLogger.debug( "Discovering URL for service", "%s -> %s" % ( self._destinationSrv, sURL ) )
     return S_OK( sURL )
-
+  
+  def __selectUrl( self, notselect, urls ):
+    """In case when multiple services are running in the same host, a new url has to be in a different host
+    Note: If we do not have different host we will use the selected url...
+    """
+    
+    url = None
+    for i in urls:
+      retVal = Network.splitURL( i )
+      if retVal['OK']:
+        if retVal['Value'][1] != notselect[1]:  # the hots are different
+          url = i
+          break
```

Comment:
then we return None. It means no URL found. But I would say it is better to print the error message...